mozjs-sys/build.rs

```diff
@@ -308,6 +329,17 @@ fn build_spidermonkey(build_dir: &Path) {
         .status()
         .expect(&format!("Failed to run `{:?}`", make));
     assert!(result.success());
+    if target.contains("windows") {
+        let mut make_static = cc::Build::new();
+        for obj in std::fs::read_to_string(build_dir.join("js/src/build/js_static_lib.list"))
+            .unwrap()  // panic on possible file-reading errors
+            .lines()  // split the string into an iterator of string slices
+            .map(String::from) {
+            make_static.object(obj);
+        }
+        make_static.static_flag(true);
+        make_static.compile("js_static");
```

Comment:
```suggestion
        make_static.static_flag(true);
        make_static.out_dir(build_dir.join("js/src/build"));
        make_static.compile("js_static");
```

It worked with this change! But artifact becomes [super huge](https://github.com/wusyong/mozjs/actions/runs/9905684451). I'll see if I can simplify the compressed tar ball on Windows.

Reply:
We can remove most stuff from artifact anyway and only keep what we keep on linux.

The reason for size inflation is that cc-rs makes two identical-sized libs (same thing also happens with jsglue and jsapi):
![slika](https://github.com/user-attachments/assets/e73440f6-9f98-4130-9b5b-db3335b45e46)

Reply:
> I'll see if I can simplify the compressed tar ball on Windows.

Okay, in the mean time I will continue to work on other things, send PR to my fork when you are done (although we could land this separately, but then I would have to rebase ...)

Reply:
Opened https://github.com/sagudev/mozjs/pull/2

Reply:
Tnx, I did squashed all relevant commits into one: https://github.com/servo/mozjs/pull/474/commits/1a084f0738914385b89eedb779f0b3cd3c3f7ead (authorship was preserved).